src/qunit.cpp

```diff
@@ -105,16 +105,26 @@ std::map<QInterfacePtr, bitLenInt> QUnit::Cohere(std::vector<QInterfacePtr> toCo
     return ret;
 }
 
-void QUnit::Detach(bitLenInt start, bitLenInt length, QInterfacePtr dest)
+void QUnit::Detach(bitLenInt start, bitLenInt length, bool keepBits, QInterfacePtr dest)
 {
     /* TODO: This method should compose the bits for the destination without cohering the length first */
 
-    EntangleRange(start, length);
-    OrderContiguous(shards[start].unit);
+    if (length > 1) {
+        EntangleRange(start, length);
+        OrderContiguous(shards[start].unit);
+    }
 
     QInterfacePtr unit = shards[start].unit;
     bitLenInt mapped = shards[start].mapped;
 
+    if (keepBits) {
```

Comment:
I am very confused about keepBits vs nullptr dest, let's talk later

Reply:
We'll talk later, but the point is that you used nullptr dest to indicate Dispose. This isn't a dispose or a Decohere. If keepBits is true, it indicates that this is an intra-unit breakdown. If keepBits is false, it indicates that this an inter-unit breakdown. I thought about how dest could indicate the difference, but I don't see a way it can on its own. However, the logic is almost the same for intra- and inter- breakdown, so we use the same method. If you want to restyle it, please do.